--- origsrc/qmmp-0.2.3/src/ui/symboldisplay.cpp	2008-07-22 06:10:52.000000000 -0500
+++ src/qmmp-0.2.3/src/ui/symboldisplay.cpp	2009-02-24 22:25:13.982375000 -0600
@@ -32,7 +32,7 @@ SymbolDisplay::SymbolDisplay ( QWidget *
     connect ( m_skin, SIGNAL ( skinChanged() ), this, SLOT (draw()));
     draw();
     for (int i=0; i<m_digits; ++i)
-#ifdef Q_OS_FREEBSD
+#if defined(Q_OS_FREEBSD) || defined(Q_OS_CYGWIN)
         m_max += 9 * (int) pow(10,i);
 #else
         m_max += 9 * (int) exp10(i);
