--- origsrc/qmmp-0.2.3/src/plugins/CommandLineOptions/IncDecVolumeOption/CMakeLists.txt	2008-09-02 13:29:12.000000000 -0500
+++ src/qmmp-0.2.3/src/plugins/CommandLineOptions/IncDecVolumeOption/CMakeLists.txt	2009-02-24 22:10:46.310500000 -0600
@@ -45,7 +45,7 @@ QT4_WRAP_CPP(libincdecvolumeoption_MOC_S
 # the UI file won't be wrapped!
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
-ADD_LIBRARY(incdecvolumeoption SHARED ${libincdecvolumeoption_SRCS} ${libincdecvolumeoption_MOC_SRCS} 
+ADD_LIBRARY(incdecvolumeoption MODULE ${libincdecvolumeoption_SRCS} ${libincdecvolumeoption_MOC_SRCS} 
            ${libincdecvolumeoption_RCC_SRCS})
 add_dependencies(incdecvolumeoption qmmpui)
 target_link_libraries(incdecvolumeoption ${QT_LIBRARIES} -lqmmpui)
--- origsrc/qmmp-0.2.3/src/plugins/Effect/srconverter/CMakeLists.txt	2008-09-02 13:29:12.000000000 -0500
+++ src/qmmp-0.2.3/src/plugins/Effect/srconverter/CMakeLists.txt	2009-02-24 22:10:46.326125000 -0600
@@ -61,7 +61,7 @@ QT4_WRAP_UI(libsrconverter_UIS_H ${libsr
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
 IF(SAMPLERATE_FOUND)
-ADD_LIBRARY(srconverter SHARED ${libsrconverter_SRCS} ${libsrconverter_MOC_SRCS} ${libsrconverter_UIS_H}
+ADD_LIBRARY(srconverter MODULE ${libsrconverter_SRCS} ${libsrconverter_MOC_SRCS} ${libsrconverter_UIS_H}
   ${libsrconverter_RCC_SRCS})
 add_dependencies(srconverter qmmp)
 target_link_libraries(srconverter ${QT_LIBRARIES} -lqmmp ${SAMPLERATE_LDFLAGS} ${SAMPLERATE_CFLAGS})
--- origsrc/qmmp-0.2.3/src/plugins/FileDialogs/QmmpFileDialog/CMakeLists.txt	2008-09-02 13:29:12.000000000 -0500
+++ src/qmmp-0.2.3/src/plugins/FileDialogs/QmmpFileDialog/CMakeLists.txt	2009-02-24 22:10:46.341750000 -0600
@@ -55,7 +55,7 @@ QT4_WRAP_UI(libqmmpfiledialog_UIS_H ${li
 # the UI file won't be wrapped!
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
-ADD_LIBRARY(qmmpfiledialog SHARED ${libqmmpfiledialog_SRCS} ${libqmmpfiledialog_MOC_SRCS} ${libqmmpfiledialog_UIS_H} ${libqmmpfiledialog_RCC_SRCS})
+ADD_LIBRARY(qmmpfiledialog MODULE ${libqmmpfiledialog_SRCS} ${libqmmpfiledialog_MOC_SRCS} ${libqmmpfiledialog_UIS_H} ${libqmmpfiledialog_RCC_SRCS})
 add_dependencies(qmmpfiledialog qmmpui)
 target_link_libraries(qmmpfiledialog ${QT_LIBRARIES} -lqmmpui)
 install(TARGETS qmmpfiledialog DESTINATION ${LIB_DIR}/qmmp/FileDialogs)
--- origsrc/qmmp-0.2.3/src/plugins/General/dbuscontrol/CMakeLists.txt	2008-09-02 13:29:12.000000000 -0500
+++ src/qmmp-0.2.3/src/plugins/General/dbuscontrol/CMakeLists.txt	2009-02-24 22:10:46.357375000 -0600
@@ -54,7 +54,7 @@ QT4_WRAP_CPP(libdbuscontrol_MOC_SRCS ${l
 # the UI file won't be wrapped!
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
-ADD_LIBRARY(dbuscontrol SHARED ${libdbuscontrol_SRCS} ${libdbuscontrol_MOC_SRCS} ${libdbuscontrol_RCC_SRCS})
+ADD_LIBRARY(dbuscontrol MODULE ${libdbuscontrol_SRCS} ${libdbuscontrol_MOC_SRCS} ${libdbuscontrol_RCC_SRCS})
 add_dependencies(dbuscontrol qmmpui)
 target_link_libraries(dbuscontrol ${QT_LIBRARIES} -lqmmpui)
 install(TARGETS dbuscontrol DESTINATION ${LIB_DIR}/qmmp/General)
--- origsrc/qmmp-0.2.3/src/plugins/General/notifier/CMakeLists.txt	2008-09-02 13:29:12.000000000 -0500
+++ src/qmmp-0.2.3/src/plugins/General/notifier/CMakeLists.txt	2009-02-24 22:10:46.373000000 -0600
@@ -60,7 +60,7 @@ QT4_WRAP_UI(libnotifier_UIS_H ${libnotif
 # the UI file won't be wrapped!
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
-ADD_LIBRARY(notifier SHARED ${libnotifier_SRCS} ${libnotifier_MOC_SRCS} ${libnotifier_UIS_H} ${libnotifier_RCC_SRCS})
+ADD_LIBRARY(notifier MODULE ${libnotifier_SRCS} ${libnotifier_MOC_SRCS} ${libnotifier_UIS_H} ${libnotifier_RCC_SRCS})
 add_dependencies(notifier qmmpui)
 target_link_libraries(notifier ${QT_LIBRARIES} -lqmmpui)
 install(TARGETS notifier DESTINATION ${LIB_DIR}/qmmp/General)
--- origsrc/qmmp-0.2.3/src/plugins/General/scrobbler/CMakeLists.txt	2008-09-02 13:29:12.000000000 -0500
+++ src/qmmp-0.2.3/src/plugins/General/scrobbler/CMakeLists.txt	2009-02-24 22:10:46.373000000 -0600
@@ -62,7 +62,7 @@ QT4_WRAP_UI(libscrobbler_UIS_H ${libscro
 # the UI file won't be wrapped!
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
-ADD_LIBRARY(scrobbler SHARED ${libscrobbler_SRCS} ${libscrobbler_MOC_SRCS} ${libscrobbler_UIS_H}
+ADD_LIBRARY(scrobbler MODULE ${libscrobbler_SRCS} ${libscrobbler_MOC_SRCS} ${libscrobbler_UIS_H}
   ${libscrobbler_RCC_SRCS})
 add_dependencies(scrobbler qmmpui)
 target_link_libraries(scrobbler ${QT_LIBRARIES} -lqmmpui)
--- origsrc/qmmp-0.2.3/src/plugins/General/statusicon/CMakeLists.txt	2008-09-02 13:29:12.000000000 -0500
+++ src/qmmp-0.2.3/src/plugins/General/statusicon/CMakeLists.txt	2009-02-24 22:10:46.388625000 -0600
@@ -58,7 +58,7 @@ QT4_WRAP_UI(libstatusicon_UIS_H ${libsta
 # the UI file won't be wrapped!
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
-ADD_LIBRARY(statusicon SHARED ${libstatusicon_SRCS} ${libstatusicon_MOC_SRCS} ${libstatusicon_UIS_H} ${libstatusicon_RCC_SRCS})
+ADD_LIBRARY(statusicon MODULE ${libstatusicon_SRCS} ${libstatusicon_MOC_SRCS} ${libstatusicon_UIS_H} ${libstatusicon_RCC_SRCS})
 add_dependencies(statusicon qmmpui)
 target_link_libraries(statusicon ${QT_LIBRARIES} -lqmmpui)
 install(TARGETS statusicon DESTINATION ${LIB_DIR}/qmmp/General)
--- origsrc/qmmp-0.2.3/src/plugins/Input/ffmpeg/CMakeLists.txt	2008-11-11 12:30:07.000000000 -0600
+++ src/qmmp-0.2.3/src/plugins/Input/ffmpeg/CMakeLists.txt	2009-02-24 22:10:46.404250000 -0600
@@ -98,7 +98,7 @@ QT4_WRAP_UI(libffmpeg_UIS_H ${libffmpeg_
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
 IF(FFMPEG_FOUND)
-ADD_LIBRARY(ffmpeg SHARED ${libffmpeg_SRCS} ${libffmpeg_MOC_SRCS} ${libffmpeg_UIS_H}
+ADD_LIBRARY(ffmpeg MODULE ${libffmpeg_SRCS} ${libffmpeg_MOC_SRCS} ${libffmpeg_UIS_H}
   ${libffmpeg_RCC_SRCS})
 add_dependencies(ffmpeg qmmp)
 target_link_libraries(ffmpeg ${QT_LIBRARIES} -lqmmp ${FFMPEG_LDFLAGS} ${FFMPEG_CFLAGS})
--- origsrc/qmmp-0.2.3/src/plugins/Input/flac/CMakeLists.txt	2008-09-02 13:29:12.000000000 -0500
+++ src/qmmp-0.2.3/src/plugins/Input/flac/CMakeLists.txt	2009-02-24 22:10:46.419875000 -0600
@@ -67,7 +67,7 @@ QT4_WRAP_UI(libflac_UIS_H ${libflac_UIS}
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
 IF(FLAC_FOUND)
-ADD_LIBRARY(flac SHARED ${libflac_SRCS} ${libflac_MOC_SRCS} ${libflac_UIS_H} ${libflac_RCC_SRCS})
+ADD_LIBRARY(flac MODULE ${libflac_SRCS} ${libflac_MOC_SRCS} ${libflac_UIS_H} ${libflac_RCC_SRCS})
 add_dependencies(flac qmmp)
 target_link_libraries(flac ${QT_LIBRARIES} -lqmmp ${FLAC_LDFLAGS} ${FLAC_CFLAGS} ${TAGLIB_LDFLAGS} ${TAGLIB_CFLAGS})
 install(TARGETS flac DESTINATION ${LIB_DIR}/qmmp/Input)
--- origsrc/qmmp-0.2.3/src/plugins/Input/mad/CMakeLists.txt	2008-09-02 13:29:12.000000000 -0500
+++ src/qmmp-0.2.3/src/plugins/Input/mad/CMakeLists.txt	2009-02-24 22:10:46.419875000 -0600
@@ -68,7 +68,7 @@ QT4_WRAP_UI(libmad_UIS_H ${libmad_UIS})
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
 IF(MAD_FOUND)
-ADD_LIBRARY(mad SHARED ${libmad_SRCS} ${libmad_MOC_SRCS} ${libmad_UIS_H} ${libmad_RCC_SRCS})
+ADD_LIBRARY(mad MODULE ${libmad_SRCS} ${libmad_MOC_SRCS} ${libmad_UIS_H} ${libmad_RCC_SRCS})
 add_dependencies(mad qmmp)
 target_link_libraries(mad ${QT_LIBRARIES} -lqmmp ${MAD_LDFLAGS} ${MAD_CFLAGS} ${TAGLIB_LDFLAGS} ${TAGLIB_CFLAGS})
 install(TARGETS mad DESTINATION ${LIB_DIR}/qmmp/Input)
--- origsrc/qmmp-0.2.3/src/plugins/Input/modplug/CMakeLists.txt	2008-11-12 12:50:39.000000000 -0600
+++ src/qmmp-0.2.3/src/plugins/Input/modplug/CMakeLists.txt	2009-02-24 22:10:46.435500000 -0600
@@ -81,7 +81,7 @@ QT4_WRAP_UI(libmodplug_UIS_H ${libmodplu
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
 IF(MODPLUG_FOUND)
-ADD_LIBRARY(modplug SHARED ${libmodplug_SRCS} ${libmodplug_MOC_SRCS} ${libmodplug_UIS_H}
+ADD_LIBRARY(modplug MODULE ${libmodplug_SRCS} ${libmodplug_MOC_SRCS} ${libmodplug_UIS_H}
   ${libmodplug_RCC_SRCS})
 add_dependencies(modplug qmmp)
 target_link_libraries(modplug ${QT_LIBRARIES} -lqmmp ${MODPLUG_LDFLAGS} ${MODPLUG_CFLAGS})
--- origsrc/qmmp-0.2.3/src/plugins/Input/mpc/CMakeLists.txt	2008-09-02 13:29:12.000000000 -0500
+++ src/qmmp-0.2.3/src/plugins/Input/mpc/CMakeLists.txt	2009-02-24 22:10:46.451125000 -0600
@@ -65,7 +65,7 @@ QT4_WRAP_UI(libmpc_UIS_H ${libmpc_UIS})
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
 IF(MPC_FOUND)
-ADD_LIBRARY(mpc SHARED ${libmpc_SRCS} ${libmpc_MOC_SRCS} ${libmpc_UIS_H}
+ADD_LIBRARY(mpc MODULE ${libmpc_SRCS} ${libmpc_MOC_SRCS} ${libmpc_UIS_H}
   ${libmpc_RCC_SRCS})
 add_dependencies(mpc qmmp)
 target_link_libraries(mpc ${QT_LIBRARIES} -lqmmp -lmpcdec ${TAGLIB_LDFLAGS} ${TAGLIB_CFLAGS})
--- origsrc/qmmp-0.2.3/src/plugins/Input/sndfile/CMakeLists.txt	2008-09-02 13:29:12.000000000 -0500
+++ src/qmmp-0.2.3/src/plugins/Input/sndfile/CMakeLists.txt	2009-02-24 22:10:46.451125000 -0600
@@ -53,7 +53,7 @@ QT4_WRAP_CPP(libsndfile_MOC_SRCS ${libsn
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
 IF(SNDFILE_FOUND)
-ADD_LIBRARY(sndfile SHARED ${libsndfile_SRCS} ${libsndfile_MOC_SRCS} ${libsndfile_UIS_H}
+ADD_LIBRARY(sndfile MODULE ${libsndfile_SRCS} ${libsndfile_MOC_SRCS} ${libsndfile_UIS_H}
   ${libsndfile_RCC_SRCS})
 add_dependencies(sndfile qmmp)
 target_link_libraries(sndfile ${QT_LIBRARIES} -lqmmp ${SNDFILE_LDFLAGS} ${SNDFILE_CFLAGS})
--- origsrc/qmmp-0.2.3/src/plugins/Input/vorbis/CMakeLists.txt	2008-09-02 13:29:12.000000000 -0500
+++ src/qmmp-0.2.3/src/plugins/Input/vorbis/CMakeLists.txt	2009-02-24 22:10:46.466750000 -0600
@@ -62,7 +62,7 @@ QT4_WRAP_UI(libvorbis_UIS_H ${libvorbis_
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
 IF(VORBIS_FOUND)
-ADD_LIBRARY(vorbis SHARED ${libvorbis_SRCS} ${libvorbis_MOC_SRCS} ${libvorbis_UIS_H}
+ADD_LIBRARY(vorbis MODULE ${libvorbis_SRCS} ${libvorbis_MOC_SRCS} ${libvorbis_UIS_H}
   ${libvorbis_RCC_SRCS})
 add_dependencies(vorbis qmmp)
 target_link_libraries(vorbis ${QT_LIBRARIES} -lqmmp ${VORBIS_LDFLAGS} ${VORBIS_CFLAGS} ${TAGLIB_LDFLAGS} ${TAGLIB_CFLAGS})
--- origsrc/qmmp-0.2.3/src/plugins/Input/wavpack/CMakeLists.txt	2008-09-02 13:29:12.000000000 -0500
+++ src/qmmp-0.2.3/src/plugins/Input/wavpack/CMakeLists.txt	2009-02-24 22:10:46.482375000 -0600
@@ -62,7 +62,7 @@ QT4_WRAP_UI(libwavpack_UIS_H ${libwavpac
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
 IF(WAVPACK_FOUND)
-ADD_LIBRARY(wavpack SHARED ${libwavpack_SRCS} ${libwavpack_MOC_SRCS} ${libwavpack_UIS_H}
+ADD_LIBRARY(wavpack MODULE ${libwavpack_SRCS} ${libwavpack_MOC_SRCS} ${libwavpack_UIS_H}
   ${libwavpack_RCC_SRCS})
 add_dependencies(wavpack qmmp)
 target_link_libraries(wavpack ${QT_LIBRARIES} -lqmmp ${WAVPACK_LDFLAGS} ${WAVPACK_CLAGS})
--- origsrc/qmmp-0.2.3/src/plugins/Output/alsa/CMakeLists.txt	2008-09-02 13:29:12.000000000 -0500
+++ src/qmmp-0.2.3/src/plugins/Output/alsa/CMakeLists.txt	2009-02-24 22:10:46.482375000 -0600
@@ -61,7 +61,7 @@ QT4_WRAP_UI(libalsa_UIS_H ${libalsa_UIS}
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
 IF(ALSA_FOUND)
-ADD_LIBRARY(alsa SHARED ${libalsa_SRCS} ${libalsa_MOC_SRCS} ${libalsa_UIS_H}
+ADD_LIBRARY(alsa MODULE ${libalsa_SRCS} ${libalsa_MOC_SRCS} ${libalsa_UIS_H}
   ${libalsa_RCC_SRCS})
 add_dependencies(alsa qmmp)
 target_link_libraries(alsa ${QT_LIBRARIES} -lqmmp ${ALSA_LDFLAGS} ${ALSA_CFLAGS})
--- origsrc/qmmp-0.2.3/src/plugins/Output/jack/CMakeLists.txt	2008-09-02 13:29:12.000000000 -0500
+++ src/qmmp-0.2.3/src/plugins/Output/jack/CMakeLists.txt	2009-02-24 22:10:46.513625000 -0600
@@ -51,7 +51,7 @@ QT4_WRAP_CPP(libjack_MOC_SRCS ${libjack_
 
 
 IF(JACK_FOUND)
-ADD_LIBRARY(jack SHARED ${libjack_SRCS} ${libjack_MOC_SRCS} ${libjack_RCC_SRCS})
+ADD_LIBRARY(jack MODULE ${libjack_SRCS} ${libjack_MOC_SRCS} ${libjack_RCC_SRCS})
 add_dependencies(jack qmmp)
 target_link_libraries(jack ${QT_LIBRARIES} -lqmmp ${JACK_LDFLAGS} ${JACK_CFLAGS})
 install(TARGETS jack DESTINATION ${LIB_DIR}/qmmp/Output)
--- origsrc/qmmp-0.2.3/src/plugins/Output/oss/CMakeLists.txt	2008-12-01 17:13:46.000000000 -0600
+++ src/qmmp-0.2.3/src/plugins/Output/oss/CMakeLists.txt	2009-02-24 22:10:46.513625000 -0600
@@ -57,7 +57,7 @@ SET(liboss_UIS
 
 QT4_WRAP_UI(liboss_UIS_H ${liboss_UIS})
 
-ADD_LIBRARY(oss SHARED ${liboss_SRCS} ${liboss_MOC_SRCS} ${liboss_UIS_H}
+ADD_LIBRARY(oss MODULE ${liboss_SRCS} ${liboss_MOC_SRCS} ${liboss_UIS_H}
   ${liboss_RCC_SRCS})
 add_dependencies(oss qmmp)
 target_link_libraries(oss ${QT_LIBRARIES} -lqmmp)
--- origsrc/qmmp-0.2.3/src/plugins/Output/pulseaudio/CMakeLists.txt	2008-09-02 13:29:12.000000000 -0500
+++ src/qmmp-0.2.3/src/plugins/Output/pulseaudio/CMakeLists.txt	2009-02-24 22:10:46.529250000 -0600
@@ -53,7 +53,7 @@ QT4_WRAP_CPP(libpulseaudio_MOC_SRCS ${li
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
 IF(PULSE_FOUND)
-ADD_LIBRARY(pulseaudio SHARED ${libpulseaudio_SRCS} ${libpulseaudio_MOC_SRCS} ${libpulseaudio_UIS_H}
+ADD_LIBRARY(pulseaudio MODULE ${libpulseaudio_SRCS} ${libpulseaudio_MOC_SRCS} ${libpulseaudio_UIS_H}
   ${libpulseaudio_RCC_SRCS})
 add_dependencies(pulseaudio qmmp)
 target_link_libraries(pulseaudio ${QT_LIBRARIES} -lqmmp ${PULSE_LDFLAGS} ${PULSE_CFLAGS})
--- origsrc/qmmp-0.2.3/src/plugins/PlaylistFormats/m3u/CMakeLists.txt	2008-09-02 13:29:12.000000000 -0500
+++ src/qmmp-0.2.3/src/plugins/PlaylistFormats/m3u/CMakeLists.txt	2009-02-24 22:10:46.544875000 -0600
@@ -43,7 +43,7 @@ QT4_WRAP_CPP(libm3uplaylistformat_MOC_SR
 # the UI file won't be wrapped!
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
-ADD_LIBRARY(m3uplaylistformat SHARED ${libm3uplaylistformat_SRCS} ${libm3uplaylistformat_MOC_SRCS})
+ADD_LIBRARY(m3uplaylistformat MODULE ${libm3uplaylistformat_SRCS} ${libm3uplaylistformat_MOC_SRCS})
 add_dependencies(m3uplaylistformat qmmpui)
 target_link_libraries(m3uplaylistformat ${QT_LIBRARIES} -lqmmpui)
 install(TARGETS m3uplaylistformat DESTINATION ${LIB_DIR}/qmmp/PlaylistFormats)
--- origsrc/qmmp-0.2.3/src/plugins/PlaylistFormats/pls/CMakeLists.txt	2008-09-02 13:29:12.000000000 -0500
+++ src/qmmp-0.2.3/src/plugins/PlaylistFormats/pls/CMakeLists.txt	2009-02-24 22:10:46.544875000 -0600
@@ -43,7 +43,7 @@ QT4_WRAP_CPP(libplsplaylistformat_MOC_SR
 # the UI file won't be wrapped!
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
-ADD_LIBRARY(plsplaylistformat SHARED ${libplsplaylistformat_SRCS} ${libplsplaylistformat_MOC_SRCS})
+ADD_LIBRARY(plsplaylistformat MODULE ${libplsplaylistformat_SRCS} ${libplsplaylistformat_MOC_SRCS})
 add_dependencies(plsplaylistformat qmmpui)
 target_link_libraries(plsplaylistformat ${QT_LIBRARIES} -lqmmpui)
 install(TARGETS plsplaylistformat DESTINATION ${LIB_DIR}/qmmp/PlaylistFormats)
--- origsrc/qmmp-0.2.3/src/plugins/PlaylistFormats/xspf/CMakeLists.txt	2008-09-02 13:29:12.000000000 -0500
+++ src/qmmp-0.2.3/src/plugins/PlaylistFormats/xspf/CMakeLists.txt	2009-02-24 22:10:46.560500000 -0600
@@ -47,7 +47,7 @@ QT4_WRAP_CPP(libxspfplaylistformat_MOC_S
 # the UI file won't be wrapped!
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
-ADD_LIBRARY(xspfplaylistformat SHARED ${libxspfplaylistformat_SRCS} ${libxspfplaylistformat_MOC_SRCS})
+ADD_LIBRARY(xspfplaylistformat MODULE ${libxspfplaylistformat_SRCS} ${libxspfplaylistformat_MOC_SRCS})
 add_dependencies(xspfplaylistformat qmmpui)
 target_link_libraries(xspfplaylistformat ${QT_LIBRARIES} -lqmmpui)
 install(TARGETS xspfplaylistformat DESTINATION ${LIB_DIR}/qmmp/PlaylistFormats)
--- origsrc/qmmp-0.2.3/src/plugins/Visual/analyzer/CMakeLists.txt	2008-09-02 13:29:12.000000000 -0500
+++ src/qmmp-0.2.3/src/plugins/Visual/analyzer/CMakeLists.txt	2009-02-24 22:10:46.576125000 -0600
@@ -60,7 +60,7 @@ QT4_WRAP_UI(libanalyzer_UIS_H ${libanaly
 # the UI file won't be wrapped!
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
-ADD_LIBRARY(analyzer SHARED ${libanalyzer_SRCS} ${libanalyzer_MOC_SRCS} ${libanalyzer_UIS_H}
+ADD_LIBRARY(analyzer MODULE ${libanalyzer_SRCS} ${libanalyzer_MOC_SRCS} ${libanalyzer_UIS_H}
   ${libanalyzer_RCC_SRCS})
 add_dependencies(analyzer qmmp)
 target_link_libraries(analyzer ${QT_LIBRARIES} -lqmmp)
--- origsrc/qmmp-0.2.3/src/qmmp/CMakeLists.txt	2008-09-01 07:57:52.000000000 -0500
+++ src/qmmp-0.2.3/src/qmmp/CMakeLists.txt	2009-02-24 22:12:37.638625000 -0600
@@ -89,5 +89,7 @@ include_directories(${CMAKE_CURRENT_BINA
 ADD_LIBRARY(libqmmp SHARED ${libqmmp_SRCS} ${libqmmp_MOC_SRCS})
 target_link_libraries(libqmmp ${QT_LIBRARIES} ${CURL_LDFLAGS} ${CURL_CFLAGS})
 SET_TARGET_PROPERTIES(libqmmp PROPERTIES VERSION ${QMMP_VERSION} SOVERSION ${QMMP_SOVERSION} OUTPUT_NAME qmmp)
-install(TARGETS libqmmp LIBRARY DESTINATION ${LIB_DIR})
+install(TARGETS libqmmp LIBRARY DESTINATION ${LIB_DIR}
+                        RUNTIME DESTINATION bin
+                        ARCHIVE DESTINATION ${LIB_DIR})
 install(FILES ${libqmmp_DEVEL_HDRS} DESTINATION include/qmmp)
--- origsrc/qmmp-0.2.3/src/qmmpui/CMakeLists.txt	2008-09-01 07:57:52.000000000 -0500
+++ src/qmmp-0.2.3/src/qmmpui/CMakeLists.txt	2009-02-24 22:12:41.841750000 -0600
@@ -72,5 +72,7 @@ include_directories(${CMAKE_CURRENT_BINA
 ADD_LIBRARY(qmmpui SHARED ${libqmmpui_SRCS} ${libqmmpui_MOC_SRCS} ${libqmmpui_RCC_SRCS})
 target_link_libraries(qmmpui ${QT_LIBRARIES})
 SET_TARGET_PROPERTIES(qmmpui PROPERTIES VERSION ${QMMP_VERSION} SOVERSION ${QMMP_SOVERSION})
-install(TARGETS qmmpui LIBRARY DESTINATION ${LIB_DIR})
+install(TARGETS qmmpui LIBRARY DESTINATION ${LIB_DIR}
+                       RUNTIME DESTINATION bin
+                       ARCHIVE DESTINATION ${LIB_DIR})
 install(FILES ${libqmmpui_DEVEL_HDRS} DESTINATION include/qmmpui)
