--- origsrc/qmmp-0.7.2/src/plugins/Output/waveout/outputwaveout.h	2013-08-26 13:02:12.000000000 -0500
+++ src/qmmp-0.7.2/src/plugins/Output/waveout/outputwaveout.h	2013-09-24 20:45:19.495153000 -0500
@@ -22,9 +22,9 @@
 #define OUTPUTWAVEOUT_H
 
 #include <stdio.h>
-#include <windows.h>
 #include <qmmp/volume.h>
 #include <qmmp/output.h>
+#include <windows.h>
 
 /**
     @author Ilya Kotov <forkotov02@hotmail.ru>
--- origsrc/qmmp-0.7.2/src/plugins/Output/waveout/outputwaveout.cpp	2013-08-26 13:02:12.000000000 -0500
+++ src/qmmp-0.7.2/src/plugins/Output/waveout/outputwaveout.cpp	2013-09-24 21:43:12.487796800 -0500
@@ -38,7 +38,7 @@ static WAVEHDR*          PlayedWaveHeade
 
 
 
-static void CALLBACK wave_callback (HWAVE hWave, UINT uMsg, DWORD dwInstance, DWORD dwParam1, DWORD dwParam2)
+static void CALLBACK wave_callback (HWAVEOUT hWave, UINT uMsg, DWORD_PTR dwInstance, DWORD_PTR dwParam1, DWORD_PTR dwParam2)
 {
     if (uMsg == WOM_DONE)
     {
@@ -98,7 +98,7 @@ bool OutputWaveOut::initialize(quint32 f
     fmt.nBlockAlign     = fmt.nChannels * fmt.wBitsPerSample/8;
     fmt.nAvgBytesPerSec = fmt.nSamplesPerSec * fmt.nChannels * fmt.wBitsPerSample/8;
 
-    switch (waveOutOpen (&dev, deviceID, &fmt, (DWORD)wave_callback, 0, CALLBACK_FUNCTION))
+    switch (waveOutOpen (&dev, deviceID, &fmt, (DWORD_PTR)wave_callback, 0, CALLBACK_FUNCTION))
     {
     case MMSYSERR_ALLOCATED:
         qWarning("OutputWaveOut: Device is already open.");
